### PR TITLE
Add TypeScript definition

### PR DESCRIPTION
**Pull request by BendingBender**
I'm not sure, whether it's a good idea to define types in `index.d.ts` while the main JS is in `src/screenfull.js`.

I'd actually prefer to put the types along the main JS file in `src/screenfull.d.ts` and extend the `grunt` config to copy this file over to `dist`.

The only problem to this is that `tsd-check` currently has two bugs preventing this. I've opened [a PR](https://github.com/SamVerschueren/tsd-check/pull/12) with a fix, so if you want me to follow this path, we will have to wait for a new release of `tsd-check`.

### CODE REVIEW

**Comment by sindresorhus** [comment]
> so if you want me to follow this path, we will have to wait for a new release of tsd-check.

👍 Sounds like the best solution. I can wait for that PR.

File: index.d.ts [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+	onerror(handler: (event: Event) => void): void;
+}
+
+/**
+ * Simple wrapper for cross-browser usage of the JavaScript [Fullscreen API](https://developer.mozilla.org/en/DOM/Using_full-screen_mode), which lets you bring the page or any element into fullscreen. Smoothens out the browser implementation differences, so you don't have to.
+ */
+declare const screenfull: Screenfull | false;
+
+export default screenfull;
+export as namespace screenfull;
```
Comment (sindresorhus) [inline index.d.ts]: Why is this needed? We already export the `Screenfull` namespace.

File: index.test-d.ts [hunk 2]
```diff
@@ -0,0 +1,32 @@
+import {expectType} from 'tsd-check';
+import screenfull from '.';
+
+if (screenfull) {
+	expectType<boolean>(screenfull.enabled);
+	expectType<boolean>(screenfull.isFullscreen);
+	expectType<Element | null>(screenfull.element);
+
+	if (screenfull.enabled) {
+		const elem: Element = document.getElementById('target')!;
```
Comment (sindresorhus) [inline index.test-d.ts]:
```suggestion
		const element: Element = document.getElementById('target')!;
```

File: index.d.ts (hunk 1 above)
Comment (BendingBender) [reply index.d.ts]: This is for non-CJS usage: https://github.com/Microsoft/TypeScript/wiki/What's-new-in-TypeScript#support-for-umd-module-definitions

File: index.test-d.ts (hunk 2 above)
Comment (BendingBender) [reply index.test-d.ts]: Nothing escapes you... 👍

**Comment by sindresorhus** [comment]
> I've opened a PR with a fix, so if you want me to follow this path, we will have to wait for a new release of tsd-check.

This is fixed now.
